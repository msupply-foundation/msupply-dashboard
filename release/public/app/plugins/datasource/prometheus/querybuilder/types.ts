import { VisualQueryBinary } from './shared/LokiAndPromQueryModellerBase';
import { QueryBuilderLabelFilter, QueryBuilderOperation } from './shared/types';

/**
 * Visual query model
 */
export interface PromVisualQuery {
  metric: string;
  labels: QueryBuilderLabelFilter[];
  operations: QueryBuilderOperation[];
  binaryQueries?: PromVisualQueryBinary[];
}

export type PromVisualQueryBinary = VisualQueryBinary<PromVisualQuery>;

export enum PromVisualQueryOperationCategory {
  Aggregations = 'Aggregations',
  RangeFunctions = 'Range functions',
  Functions = 'Functions',
  BinaryOps = 'Binary operations',
}

export interface PromQueryPattern {
  name: string;
  operations: QueryBuilderOperation[];
}

export function getDefaultEmptyQuery() {
  const model: PromVisualQuery = {
    metric: '',
    labels: [],
    operations: [],
  };

  return model;
}
