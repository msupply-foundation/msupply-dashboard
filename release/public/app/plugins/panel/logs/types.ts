import { LogsSortOrder, LogsDedupStrategy } from '@grafana/data';

export interface Options {
  showLabels: boolean;
  showTime: boolean;
  wrapLogMessage: boolean;
  enableLogDetails: boolean;
  sortOrder: LogsSortOrder;
  dedupStrategy: LogsDedupStrategy;
}
