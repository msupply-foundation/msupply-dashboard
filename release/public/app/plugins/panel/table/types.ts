export interface Options {
  frameIndex: number;
  showHeader: boolean;
}

export interface CustomFieldConfig {
  width: number;
  displayMode: string;
}
