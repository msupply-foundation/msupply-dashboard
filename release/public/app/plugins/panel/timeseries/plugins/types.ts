interface AnnotationsDataFrameViewDTO {
  time: number;
  text: string;
  tags: string[];
  alertId?: number;
  newState?: string;
  title?: string;
  color: string;
}
