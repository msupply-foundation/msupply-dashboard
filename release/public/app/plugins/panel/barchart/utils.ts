import {
  DataFrame,
  FieldType,
  formattedValueToString,
  getFieldColorModeForField,
  getFieldSeriesColor,
  MutableDataFrame,
  VizOrientation,
} from '@grafana/data';
import { BarChartFieldConfig, BarChartOptions, defaultBarChartFieldConfig } from './types';
import { BarsOptions, getConfig } from './bars';
import {
  AxisPlacement,
  FIXED_UNIT,
  ScaleDirection,
  ScaleDistribution,
  ScaleOrientation,
  UPlotConfigBuilder,
  UPlotConfigPrepFn,
} from '@grafana/ui';

/** @alpha */
function getBarCharScaleOrientation(orientation: VizOrientation) {
  if (orientation === VizOrientation.Vertical) {
    return {
      xOri: ScaleOrientation.Horizontal,
      xDir: ScaleDirection.Right,
      yOri: ScaleOrientation.Vertical,
      yDir: ScaleDirection.Up,
    };
  }

  return {
    xOri: ScaleOrientation.Vertical,
    xDir: ScaleDirection.Down,
    yOri: ScaleOrientation.Horizontal,
    yDir: ScaleDirection.Right,
  };
}

export const preparePlotConfigBuilder: UPlotConfigPrepFn<BarChartOptions> = ({
  frame,
  theme,
  orientation,
  showValue,
  groupWidth,
  barWidth,
  text,
}) => {
  const builder = new UPlotConfigBuilder();
  const defaultValueFormatter = (seriesIdx: number, value: any) =>
    formattedValueToString(frame.fields[seriesIdx].display!(value));

  // bar orientation -> x scale orientation & direction
  const vizOrientation = getBarCharScaleOrientation(orientation);

  const formatValue = defaultValueFormatter;

  // Use bar width when only one field
  if (frame.fields.length === 2) {
    groupWidth = barWidth;
    barWidth = 1;
  }

  const opts: BarsOptions = {
    xOri: vizOrientation.xOri,
    xDir: vizOrientation.xDir,
    groupWidth,
    barWidth,
    formatValue,
    text,
    showValue,
  };

  const config = getConfig(opts, theme);

  builder.addHook('init', config.init);
  builder.addHook('drawClear', config.drawClear);
  builder.addHook('draw', config.draw);

  builder.setTooltipInterpolator(config.interpolateTooltip);

  builder.addScale({
    scaleKey: 'x',
    isTime: false,
    distribution: ScaleDistribution.Ordinal,
    orientation: vizOrientation.xOri,
    direction: vizOrientation.xDir,
  });

  builder.addAxis({
    scaleKey: 'x',
    isTime: false,
    placement: vizOrientation.xOri === 0 ? AxisPlacement.Bottom : AxisPlacement.Left,
    splits: config.xSplits,
    values: config.xValues,
    grid: false,
    ticks: false,
    gap: 15,
    theme,
  });

  let seriesIndex = 0;

  // iterate the y values
  for (let i = 1; i < frame.fields.length; i++) {
    const field = frame.fields[i];

    field.state!.seriesIndex = seriesIndex++;

    const customConfig: BarChartFieldConfig = { ...defaultBarChartFieldConfig, ...field.config.custom };

    const scaleKey = field.config.unit || FIXED_UNIT;
    const colorMode = getFieldColorModeForField(field);
    const scaleColor = getFieldSeriesColor(field, theme);
    const seriesColor = scaleColor.color;

    builder.addSeries({
      scaleKey,
      pxAlign: false,
      lineWidth: customConfig.lineWidth,
      lineColor: seriesColor,
      fillOpacity: customConfig.fillOpacity,
      theme,
      colorMode,
      pathBuilder: config.drawBars,
      show: !customConfig.hideFrom?.viz,
      gradientMode: customConfig.gradientMode,
      thresholds: field.config.thresholds,

      // The following properties are not used in the uPlot config, but are utilized as transport for legend config
      dataFrameFieldIndex: {
        fieldIndex: i,
        frameIndex: 0,
      },
    });

    // The builder will manage unique scaleKeys and combine where appropriate
    builder.addScale({
      scaleKey,
      min: field.config.min,
      max: field.config.max,
      softMin: customConfig.axisSoftMin,
      softMax: customConfig.axisSoftMax,
      orientation: vizOrientation.yOri,
      direction: vizOrientation.yDir,
    });

    if (customConfig.axisPlacement !== AxisPlacement.Hidden) {
      let placement = customConfig.axisPlacement;
      if (!placement || placement === AxisPlacement.Auto) {
        placement = AxisPlacement.Left;
      }
      if (vizOrientation.xOri === 1) {
        if (placement === AxisPlacement.Left) {
          placement = AxisPlacement.Bottom;
        }
        if (placement === AxisPlacement.Right) {
          placement = AxisPlacement.Top;
        }
      }

      builder.addAxis({
        scaleKey,
        label: customConfig.axisLabel,
        size: customConfig.axisWidth,
        placement,
        formatValue: (v) => formattedValueToString(field.display!(v)),
        theme,
      });
    }
  }

  return builder;
};

/** @internal */
export function preparePlotFrame(data: DataFrame[]) {
  const firstFrame = data[0];
  const firstString = firstFrame.fields.find((f) => f.type === FieldType.string);

  if (!firstString) {
    throw new Error('No string field in DF');
  }

  const resultFrame = new MutableDataFrame();
  resultFrame.addField(firstString);

  for (const f of firstFrame.fields) {
    if (f.type === FieldType.number) {
      resultFrame.addField(f);
    }
  }

  return resultFrame;
}
