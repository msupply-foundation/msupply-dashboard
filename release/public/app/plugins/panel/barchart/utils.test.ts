import { preparePlotConfigBuilder, preparePlotFrame } from './utils';
import {
  createTheme,
  DefaultTimeZone,
  EventBusSrv,
  FieldConfig,
  FieldType,
  getDefaultTimeRange,
  MutableDataFrame,
  VizOrientation,
} from '@grafana/data';
import { BarChartFieldConfig, BarChartOptions } from './types';
import {
  BarValueVisibility,
  GraphGradientMode,
  LegendDisplayMode,
  StackingMode,
  TooltipDisplayMode,
} from '@grafana/ui';

function mockDataFrame() {
  const df1 = new MutableDataFrame({
    refId: 'A',
    fields: [{ name: 'ts', type: FieldType.string, values: ['a', 'b', 'c'] }],
  });

  const df2 = new MutableDataFrame({
    refId: 'B',
    fields: [{ name: 'ts', type: FieldType.time, values: [1, 2, 4] }],
  });

  const f1Config: FieldConfig<BarChartFieldConfig> = {
    displayName: 'Metric 1',
    decimals: 2,
    unit: 'm/s',
    custom: {
      gradientMode: GraphGradientMode.Opacity,
      lineWidth: 2,
      fillOpacity: 0.1,
    },
  };

  const f2Config: FieldConfig<BarChartFieldConfig> = {
    displayName: 'Metric 2',
    decimals: 2,
    unit: 'kWh',
    custom: {
      gradientMode: GraphGradientMode.Hue,
      lineWidth: 2,
      fillOpacity: 0.1,
    },
  };

  df1.addField({
    name: 'metric1',
    type: FieldType.number,
    config: f1Config,
    state: {},
  });

  df2.addField({
    name: 'metric2',
    type: FieldType.number,
    config: f2Config,
    state: {},
  });

  return preparePlotFrame([df1, df2]);
}

jest.mock('@grafana/data', () => ({
  ...(jest.requireActual('@grafana/data') as any),
  DefaultTimeZone: 'utc',
}));

describe('BarChart utils', () => {
  describe('preparePlotConfigBuilder', () => {
    const frame = mockDataFrame();

    const config: BarChartOptions = {
      orientation: VizOrientation.Auto,
      groupWidth: 20,
      barWidth: 2,
      showValue: BarValueVisibility.Always,
      legend: {
        displayMode: LegendDisplayMode.List,
        placement: 'bottom',
        calcs: [],
      },
      stacking: StackingMode.None,
      tooltip: {
        mode: TooltipDisplayMode.None,
      },
      text: {
        valueSize: 10,
      },
    };

    it.each([VizOrientation.Auto, VizOrientation.Horizontal, VizOrientation.Vertical])('orientation', (v) => {
      const result = preparePlotConfigBuilder({
        ...config,
        orientation: v,
        frame: frame!,
        theme: createTheme(),
        timeZone: DefaultTimeZone,
        getTimeRange: getDefaultTimeRange,
        eventBus: new EventBusSrv(),
      }).getConfig();
      expect(result).toMatchSnapshot();
    });

    it.each([BarValueVisibility.Always, BarValueVisibility.Auto])('value visibility', (v) => {
      expect(
        preparePlotConfigBuilder({
          ...config,
          showValue: v,
          frame: frame!,
          theme: createTheme(),
          timeZone: DefaultTimeZone,
          getTimeRange: getDefaultTimeRange,
          eventBus: new EventBusSrv(),
        }).getConfig()
      ).toMatchSnapshot();
    });

    it.each([StackingMode.None, StackingMode.Percent, StackingMode.Normal])('stacking', (v) => {
      expect(
        preparePlotConfigBuilder({
          ...config,
          stacking: v,
          frame: frame!,
          theme: createTheme(),
          timeZone: DefaultTimeZone,
          getTimeRange: getDefaultTimeRange,
          eventBus: new EventBusSrv(),
        }).getConfig()
      ).toMatchSnapshot();
    });
  });
});
