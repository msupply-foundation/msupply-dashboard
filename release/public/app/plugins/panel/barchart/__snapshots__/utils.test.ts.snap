// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart utils preparePlotConfigBuilder orientation 1`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {},
    Object {
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": undefined,
        "size": undefined,
        "stroke": undefined,
      },
      "pxAlign": false,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder orientation 2`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {},
    Object {
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": undefined,
        "size": undefined,
        "stroke": undefined,
      },
      "pxAlign": false,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder orientation 3`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "x",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "m/s",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": 1,
      "distr": 2,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {},
    Object {
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": undefined,
        "size": undefined,
        "stroke": undefined,
      },
      "pxAlign": false,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder stacking 1`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {},
    Object {
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": undefined,
        "size": undefined,
        "stroke": undefined,
      },
      "pxAlign": false,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder stacking 2`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {},
    Object {
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": undefined,
        "size": undefined,
        "stroke": undefined,
      },
      "pxAlign": false,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder stacking 3`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {},
    Object {
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": undefined,
        "size": undefined,
        "stroke": undefined,
      },
      "pxAlign": false,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder value visibility 1`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {},
    Object {
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": undefined,
        "size": undefined,
        "stroke": undefined,
      },
      "pxAlign": false,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder value visibility 2`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelFont": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(201, 209, 217)",
      "ticks": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {},
    Object {
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": undefined,
        "size": undefined,
        "stroke": undefined,
      },
      "pxAlign": false,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;
